### ISSUE TITLE
Possible NULL-pointer dereference

### ISSUE THREAD

**Issue by milabs** [jvinet/knock#56]
https://github.com/jvinet/knock/blob/master/src/knockd.c#L1631

~~~
	/* create a new entry */
	attempt = (knocker_t*)malloc(sizeof(knocker_t));
	attempt->srchost = NULL; // <<-- NULL-ptr deref if malloc fails
	if(attempt == NULL) {
		perror("malloc");
		exit(1);
	}
~~~

**Comment by TDFKAOlli** [comment]
Already addressed by pull request #52

**Comment by milabs** [comment]
Ah, ok... didn't mentioned it
